aptos-move/framework/supra-framework/sources/multisig_voting.spec.move

```diff
@@ -0,0 +1,427 @@
+        //TODO: Remove pragma aborts_if_is_strict;
+        // pragma aborts_if_is_strict;
+    }
+
+    spec register {
+        let addr = signer::address_of(account);
+
+        // Will abort if there's already a `VotingForum<ProposalType>` under addr
+        aborts_if exists<VotingForum<ProposalType>>(addr);
+        // Creation of 4 new event handles changes the account's `guid_creation_num`
+        aborts_if !exists<account::Account>(addr);
+        let register_account = global<account::Account>(addr);
+        aborts_if register_account.guid_creation_num + 4 >= account::MAX_GUID_CREATION_NUM;
+        aborts_if register_account.guid_creation_num + 4 > MAX_U64;
+        // `type_info::type_of()` may abort if the type parameter is not a struct
+        aborts_if !type_info::spec_is_struct<ProposalType>();
+
+        ensures exists<VotingForum<ProposalType>>(addr);
+    }
+
+    spec create_proposal {
+        use supra_framework::chain_status;
+        //TODO: Remove pragma aborts_if_is_partial;
+        pragma aborts_if_is_partial = true;
+        requires chain_status::is_operating();
+        include CreateProposalAbortsIfAndEnsures<ProposalType>{is_multi_step_proposal: false};
+        // property 1: Verify the proposal_id of the newly created proposal.
+        /// [high-level-req-1]
+        ensures result == old(global<VotingForum<ProposalType>>(voting_forum_address)).next_proposal_id;
+    }
+
+    spec create_proposal_v2 {
+        use supra_framework::chain_status;
+        //TODO: Remove pragma aborts_if_is_partial;
+        pragma aborts_if_is_partial = true;
+        requires chain_status::is_operating();
+        include CreateProposalAbortsIfAndEnsures<ProposalType>;
+        // property 1: Verify the proposal_id of the newly created proposal.
+        ensures result == old(global<VotingForum<ProposalType>>(voting_forum_address)).next_proposal_id;
+    }
+
+    spec vote {
+        use supra_framework::chain_status;
+        // Ensures existence of Timestamp
+        requires chain_status::is_operating();
+        //TODO: Remove pragma aborts_if_is_partial;
+        pragma aborts_if_is_partial = true;
+        // property 2: While voting, it ensures that only the governance module that defines ProposalType may initiate voting
+        // and that the proposal under vote exists in the specified voting forum.
+        /// [high-level-req-2]
+        aborts_if !exists<VotingForum<ProposalType>>(voting_forum_address);
+        let voting_forum = global<VotingForum<ProposalType>>(voting_forum_address);
+        let proposal = table::spec_get(voting_forum.proposals, proposal_id);
+        // Getting proposal from voting forum might fail because of non-exist id
+        aborts_if !table::spec_contains(voting_forum.proposals, proposal_id);
+        aborts_if is_voting_period_over(proposal);
+        aborts_if proposal.is_resolved;
+        aborts_if !exists<timestamp::CurrentTimeMicroseconds>(@supra_framework);
```

Comment:
@axiongsupra not sure why this `aborts_if` is required and what purpose does it serve.

Reply:
This is for checking the whether time already started. This is required if we have call to `timestamp::now_seconds()`